src/libstd/io/stdio.rs

```diff
@@ -954,15 +954,7 @@ pub fn input() -> String {
     let len = stdin().read_line(&mut input).expect("Failed to read from stdin");
     match len {
         0 => panic!("invalid character!"),
-        _ => {
-            if input.ends_with('\n') {
-                input.pop();
-                if input.ends_with('\r') {
-                    input.pop();
-                }
-            }
-            input
-        }
+        _ => String::from(input.trim_end_matches('\n').trim_end_matches('\r')),
```

Comment:
Possible alternative
```suggestion
        _ => String::from(input.trim_end_matches(&['\n', '\r'])),
```

Reply:
When i run it, it says *expected an `Fn<(char,)>` closure, found `[char; 2]`*. Have i implemented this wrong?
Link: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=cebddadd888794a2dddf9e636332c1f2

Reply:
Oh right, `Pattern` is implemented on slices, not fixed size arrays, so you get terrible diagnostics when you pass a fixed size array. Doing `&['\n','\r'][..]` seems to fix it.

Reply:
Streamlined whitespace stripping.